### PR TITLE
HRCPP-210/topology first implementation of Codec 2.0 topology

### PR DESCRIPTION
**Pull request by rigazilla**
This PR introduces data structures and code for segment topology

### CODE REVIEW

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 359](http://ci.infinispan.org/viewLog.html?buildId=37289&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for e1f1a9e3f9d9f80f663d3e4402b733d5fb73e5ab
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:02:05

Process exited with code 255

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 511](http://ci.infinispan.org/viewLog.html?buildId=37290&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for e1f1a9e3f9d9f80f663d3e4402b733d5fb73e5ab
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:07:18

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 512](http://ci.infinispan.org/viewLog.html?buildId=37291&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for e1f1a9e3f9d9f80f663d3e4402b733d5fb73e5ab
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:05:09

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 360](http://ci.infinispan.org/viewLog.html?buildId=37298&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for e1f1a9e3f9d9f80f663d3e4402b733d5fb73e5ab
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:04:37

Process exited with code 255

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 361](http://ci.infinispan.org/viewLog.html?buildId=37299&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for 0c449d3c6727dc32ec5aafb3b1287c5b84571e44
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:02:02

Process exited with code 255

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 362](http://ci.infinispan.org/viewLog.html?buildId=37302&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for b705cf05cb1c173757de52db429cd236a6934988
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:11:58

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 513](http://ci.infinispan.org/viewLog.html?buildId=37300&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for b705cf05cb1c173757de52db429cd236a6934988
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:05:19

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 514](http://ci.infinispan.org/viewLog.html?buildId=37301&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for b705cf05cb1c173757de52db429cd236a6934988
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:05:13

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 515](http://ci.infinispan.org/viewLog.html?buildId=37303&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for b705cf05cb1c173757de52db429cd236a6934988
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:05:13

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 516](http://ci.infinispan.org/viewLog.html?buildId=37304&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for b705cf05cb1c173757de52db429cd236a6934988
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:04:40

File: include/infinispan/hotrod/TimeUnit.h [hunk 1]
```diff
@@ -19,7 +19,7 @@ namespace hotrod {
         DAYS = 0x06,
         HOURS = 0x05,
         DEFAULT = 0x07,
-        INFINITE = 0x08
+        INFINITUM = 0x08
```
Comment (tristantarrant) [inline include/infinispan/hotrod/TimeUnit.h]: Latinism ? :)

**Comment by tristantarrant** [comment]
Pushed, thanks

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 368](http://ci.infinispan.org/viewLog.html?buildId=37449&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for b705cf05cb1c173757de52db429cd236a6934988
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:11:39

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 527](http://ci.infinispan.org/viewLog.html?buildId=37447&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for b705cf05cb1c173757de52db429cd236a6934988
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:05:14

**Pull request closed without merging**